llama/Makefile

```diff
@@ -278,7 +278,7 @@ DIST_RUNNERS = $(addprefix $(RUNNERS_DIST_DIR)/,$(addsuffix /ollama_llama_server
 PAYLOAD_RUNNERS = $(addprefix $(RUNNERS_PAYLOAD_DIR)/,$(addsuffix /ollama_llama_server$(EXE_EXT).gz,$(RUNNERS)))
 BUILD_RUNNERS = $(addprefix $(RUNNERS_BUILD_DIR)/,$(addsuffix /ollama_llama_server$(EXE_EXT),$(RUNNERS)))
 
-all: clean-payload dist payload 
+all: clean-payload .WAIT dist payload 
```

Comment:
This didn't fix the problem on my system but I guess there might be multiple similar races.

For me, it looks like there is a race within clean-payload itself - between rm and mkdir. Adding this solves the problem:
`.NOTPARALLEL clean-payload`

We probably still the need the `.WAIT` in all to solve the race between those dependencies.

Reply:
nit:

```makefile
.NOTPARALLEL: ...
```

Reply:
I would also suggest not depending on clean in the default target

Reply:
`clean-payload` just cleans the final build products, not everything. We need to do it so we can switch between old and new runners.

Reply:
worth noting this might not work as expected. `.NOTPARALLEL` is used to signal its prerequisites, `clean-payload` in this case, should be run sequentially. `all` will still execute _its_ prerequisites in parallel. what you probably want is to make `all` `.NOTPARALLEL`

https://www.gnu.org/software/make/manual/html_node/Special-Targets.html#index-parallel-execution_002c-overriding